src/network/networkd-dhcp6.c

```diff
@@ -35,6 +36,97 @@ static bool dhcp6_get_prefix_delegation(Link *link) {
                       RADV_PREFIX_DELEGATION_BOTH);
 }
 
+static bool dhcp6_has_preferred_subnet_id(Link *link) {
+        if (!link->network)
+                return false;
+
+        return link->network->router_prefix_subnet_id >= 0;
+}
+
+static int dhcp6_get_preferred_delegated_prefix(
+                Manager* manager,
+                Link *link,
+                const struct in6_addr *pd_prefix,
+                uint8_t pd_prefix_len,
+                struct in6_addr *ret_addr) {
+        int r;
+        union in_addr_union pd_prefix_union = {
+                .in6 = *pd_prefix,
+        };
+        const int64_t subnet_id = link->network->router_prefix_subnet_id;
```

Comment:
"const" only really makes sense on pointers really, in particular in function arguments. I'd not add the extra decoration to local plain variables. i.e. there's major benefit in declaring the contract on function prototypes, since that's one of the primary concepts that define abstraction. But for local variables? it's just noise...

(doesn't matter much though)